frameworks/kafka/tests/test_sanity.py

```diff
@@ -1,25 +1,250 @@
 import pytest
 
 import sdk_install as install
+import sdk_tasks as tasks
+import sdk_marathon as marathon
 import sdk_utils as utils
+import sdk_package as package
+import sdk_cmd as command
+import sdk_spin as spin
+import json
+import dcos
 
 from tests.config import (
     PACKAGE_NAME,
-    DEFAULT_TASK_COUNT
+    DEFAULT_BROKER_COUNT,
+    DEFAULT_PARTITION_COUNT,
+    DEFAULT_REPLICATION_FACTOR
 )
 
+from tests.test_utils import (
+    service_cli,
+    check_health,
+    get_running_broker_task
+)
+
+
+DEFAULT_TOPIC_NAME = 'topic1'
+EPHEMERAL_TOPIC_NAME = 'topic_2'
+
 
 def setup_module(module):
     install.uninstall(PACKAGE_NAME)
     utils.gc_frameworks()
-    install.install(PACKAGE_NAME, DEFAULT_TASK_COUNT)
+    install.install(PACKAGE_NAME, DEFAULT_BROKER_COUNT)
+    check_health()
+    default_topic()
 
 
 def teardown_module(module):
     install.uninstall(PACKAGE_NAME)
 
 
+# Connection
+@pytest.mark.smoke
+@pytest.mark.sanity
+def test_endpoints_address():
+    def fun():
+        return service_cli('endpoints broker')
+
+    def ends(result):
+        return (
+            len(result['native']) == DEFAULT_BROKER_COUNT,
+            'waiting for the expected number of brokers to come online'
+        )
+
+    address = spin.spin(fun, ends)
```

Comment:
I recommend using time_wait_return() here, to conform with other framework tests: https://github.com/mesosphere/dcos-commons/blob/master/testing/sdk_spin.py#L13